### ISSUE TITLE
'x-content-type-options' header is not specified

### ISSUE THREAD

**Issue by cyberh0me** [aspnet/templating#497]
-`<script src="/lib/bootstrap/dist/js/bootstrap.js"></script>`
-`<link rel="stylesheet" href="/css/site.css">`
-`<link rel="stylesheet" href="/lib/bootstrap/dist/css/bootstrap.css">`
-`<script src="/lib/jquery/dist/jquery.js"></script>`
-`<script src="/js/site.js?v=ji3-IxbEzYWjzzLCGkF1KDjrT2jLbbrSYXw-AhMPNIA"></script>`

[Require `X-Content-Type-Options` HTTP response header](https://sonarwhal.com/docs/user-guide/rules/rule-x-content-type-options/)

**Comment by blowdart** [comment]
As a work around, you can do it right now via middleware with

```c#
public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
{
	app.Use(async (context, next) =>
	{
		context.Response.Headers.Add("X-Content-Type-Options", "nosniff");
		await next();
	});
 
	app.UseMvc();
}
```

Or via [nwebsec](https://docs.nwebsec.com/en/latest/), which also includes a bunch of security headers. 

This may well call for a better header middleware in basic middleware so some of the more common security headers, without undercutting nwebsec.

**Comment by mkArtakMSFT** [comment]
So the work here will be to update the templates to consume the new middleware (referenced https://github.com/aspnet/BasicMiddleware/issues/323).

**Comment by ryanbrandenburg** [comment]
@mkArtakMSFT can we triage aspnet/BasicMiddleware#323 so we have a better idea of when we'll be able to do this?

**Comment by ryanbrandenburg** [comment]
@mkArtakMSFT still can't move on this item until aspnet/BasicMiddleware#323 is done, can we triage it?

**Comment by mkArtakMSFT** [comment]
yeah, that one will most probably land in 3.0. So pushing this out.
